mockgcp/mockserviceusage/serviceusagev1.go

```diff
@@ -273,6 +275,47 @@ func (s *ServiceUsageV1) ListServices(ctx context.Context, req *pb.ListServicesR
 		})
 	}
 
+	// Implement field filtering
+	// TODO: Can we mame this generic?  Is it used in other APIs?
+	{
+		md, _ := metadata.FromIncomingContext(ctx)
+		query := ""
+		if md != nil {
+			for _, v := range md.Get("query") {
+				query = v
```

Comment:
If query shows up more than once, it might be nice to get a warning. Especially as all but the last will be ignored.